main/src/ca/uwaterloo/flix/language/phase/MonoDefs.scala

```diff
@@ -686,6 +731,10 @@ object MonoDefs {
   /**
     * Performs type erasure on the given type `tpe`.
     *
+    * Returns the input type as-is if the input type is normalized.
+    *
+    * TODO should be deleted
+    *
     * Flix does not erase normal types, but it does erase Boolean and caseset formulas.
     */
   private def eraseType(tpe: Type)(implicit root: LoweredAst.Root, flix: Flix): Type = tpe match {
```

Comment:
Based on our discussion of typematch - if we want typematch to work always, we cannot have this function - so it seems we should delete this. It also doesn't erase things properly because it only looks at variables. It seems like a no-op to me

Reply:
Agreed it. We should not erase here.

Reply:
Rebase and its gone.